dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -460,17 +461,23 @@ def find_counterfactuals(self, query_instance, desired_range, desired_class,
 
             # rest of the next generation obtained from top 50% of fittest members of current generation
             rest_members = self.population_size - top_members
-            new_generation_2 = np.zeros((rest_members, self.data_interface.number_of_features))
-            for new_gen_idx in range(rest_members):
-                parent1 = random.choice(population[:int(len(population) / 2)])
-                parent2 = random.choice(population[:int(len(population) / 2)])
-                child = self.mate(parent1, parent2, features_to_vary, query_instance)
-                new_generation_2[new_gen_idx] = child
-
-            if self.total_CFs > 0:
-                population = np.concatenate([new_generation_1, new_generation_2])
+            new_generation_2 = None
+            if rest_members > 0:
+                new_generation_2 = np.zeros((rest_members, self.data_interface.number_of_features))
+                for new_gen_idx in range(rest_members):
+                    parent1 = random.choice(population[:int(len(population) / 2)])
+                    parent2 = random.choice(population[:int(len(population) / 2)])
+                    child = self.mate(parent1, parent2, features_to_vary, query_instance)
+                    new_generation_2[new_gen_idx] = child
+
+            if new_generation_2 is not None:
+                if self.total_CFs > 0:
+                    population = np.concatenate([new_generation_1, new_generation_2])
+                else:
+                    population = new_generation_2
             else:
-                population = new_generation_2
+                raise SystemError("The number of total_Cfs is greater than the population size!")
+                population = new_generation_1
```

Comment:
Yup, makes sense. Thanks!

Reply:
@gaugup I've addressed it in the updated PR. I was wondering about this comment

> > Missing tests. Could you please add the relevant tests?
> 
> In the current code, there is no way that the error would be raised. I've just included it in case future code changes the genetic algorithm logic. In a way, that error in itself is like a test for future code. I'm not sure how to test this logic

Reply:
Sure test on number of counterfactuals (specially large value) maybe flaky. It is fine not to add it.